common/beerocks/bwl/dwpal/mon_wlan_hal_dwpal.cpp

```diff
@@ -39,6 +44,38 @@ namespace dwpal {
             delete[] obj;                                                                          \
     })
 
+// Helper macro for dwpal_nl_cmd_get
+// interfaceName, out_buffer, out_buffer_size, max_buffer_size, paramsObj
+#define DWPAL_GET_SCAN_PARAMS_HELPER_FG(IFNAME, OUT_BUFF, OUT_BUFF_SZ, MAX_BUFF_SZ, PRMS)          \
+    if (dwpal_nl_cmd_get(IFNAME, LTQ_NL80211_VENDOR_SUBCMD_GET_SCAN_PARAMS, OUT_BUFF,              \
+                         &OUT_BUFF_SZ, MAX_BUFF_SZ) != true) {                                     \
+        LOG(ERROR) << __func__ << " LTQ_NL80211_VENDOR_SUBCMD_GET_SCAN_PARAMS failed!";            \
+        return false;                                                                              \
+    }                                                                                              \
+    memcpy_s(&PRMS, sizeof(PRMS), &OUT_BUFF[NL_ATTR_HDR], sizeof(PRMS));
```

Comment:
This is crazy - dwpal_nl_cmd_get already does a copy, and then immediately afterwards we do another copy... Just pass the params structure directly as argument to dwpal_nl_cmd_get.